### ISSUE TITLE
Update rtl88x2bu driver repository

### ISSUE THREAD

**Issue by WolfangAukang** [NixOS/nixpkgs#147053]
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version: 5.10.79-unstable-2021-05-18
desired version: unstable-2021-11-22

###### Notify maintainers

maintainers: @Ralith 

###### Note for maintainers

[This](https://github.com/morrownr/88x2bu-20210702) is the new repo for this driver, as stated on the [original repo](https://github.com/morrownr/88x2bu#readme):

News: A new version of this driver is now available. Testing has shown the new driver to be a very solid driver in managed and master modes You are invited to help test and break in the new driver:

[88x2bu-20210702](https://github.com/morrownr/88x2bu-20210702)

The plan is to discontinue this driver by the end of December 2021. This URL will then be used to provide a link to the new driver so this URL is a good permanent bookmark. Please transition to the new driver as you are able to do so. Regards. Nick

**Comment by Ralith** [comment]
I no longer have a rtl88x2bu device, so I can't test this.

**Comment by exzombie** [comment]
I have the "pleasure" of having to deal with an adapter that needs this driver (Edimax EW-7822UAD). It is supported by neither the original repo nor the updated repo (despite the fact that the README claims support). I guess it's just a matter of adding USB IDs, and I can open the issue on the updated repo.

But here's the thing. I see that there are a number of different repos with this driver in different states of maintenance. What made you pick morrownr's repo over others? I am currently using [RinCat's repo](https://github.com/RinCat/RTL88x2BU-Linux-Driver) which claims to support more hardware, and it seems to work well, though I have only tested AP mode and haven't run a benchmark for an extended period of time. Would it make sense to switch repos entirely, or would you prefer to stick to morrownr?

**Comment by WolfangAukang** [comment]
@exzombie tbh, I created this request based on someone who needed help in configuring their device, and found that notice on the repo the current package references.

Another option might be having both (not sure how this would be handled though), but I wouldn't switch repos so drastically to avoid issues for people that currently use this driver (they might not, but rather avoid future complaints).

**Comment by exzombie** [comment]
Yeah, there's that, although AFAIK all the repos have the same origin in a driver that Realtek threw out there and let to rot and I'm not sure whether any functionality was added. But there's also the question of trust and continuity. One of the repos that is the top hit on duckduckgo has gone unmaintained and won't build against new kernels. Thus, my question was directed more at @Ralith as the maintainer; perhaps there was a reason why that particular repo was chosen. Or there wasn't :) . In either case, it would be nice to know.

_Sent from my Moto X (4) using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

**Comment by Ralith** [comment]
At the time it would have been the most actively maintained repo I could find/get working.

**Comment by Prillan** [comment]
I have a AC53 Nano that I can use for testing. But I don't know what I'm supposed to test, just daily use or any particular features? Any test suites?

**Comment by exzombie** [comment]
I was considering a PR to switch to RinCat's repo on the grounds that it supports more hardware. So I've been testing it for a couple of weeks now.

However, I see now that morrownr's repo recently expanded HW support. And I agree with @WolfangAukang that the probability of regressions is lower if we maintain continuity and stick to the same upstream. To wit, I'm now testing with the following overlay:
```
  nixpkgs.overlays = [
    (self: super: {
      linuxPackages = super.linuxPackages.extend (lpself: lpsuper: {
        rtl88x2bu = super.linuxPackages.rtl88x2bu.overrideAttrs (oldAttrs: {
          src = pkgs.fetchFromGitHub {
            owner = "exzombie";
            repo = "88x2bu-20210702";
            rev = "3b1e6666aefbd1ffa4ca688a3015fd610ee6455e";
            sha256 = "sha256-SOktR9e0EjvTWefCFpb1ox5k5hy7n35bFdr4S20ZFco=";
          };
        });
      });
    })
  ];
```

It points to my personal repo where I added a USB ID for my particular adaptor. I plan to merge that upstream after testing.

@Prillan As I understand it, these USB adapters are only good for casual use anyway, so I don't think it makes sense to test more than that. What I do is:
- use the adaptor with my home wifi network
- suspend the computer every day to ensure the adaptor wakes up properly
- test with iperf occasionally (using another machine with a wired interface as the iperf server) to check the bandwidth as I've had bad experiences with other rtl adaptors in the past becoming flaky after a while
- I would also test with NetworkManager in coffee shops and such to see if network switching works well, but I can't because I use this adapter with my desktop machine.

**Comment by cryo28** [comment]
It seems that after this change the driver no longer builds for aarch64 for all available kernel versions: 5.15, 5.16, 5.17 aarch64 builds all started to fail.

cc @Ralith
